aiohttp/web.py

```diff
@@ -462,19 +686,15 @@ def run_app(
     loop: Optional[asyncio.AbstractEventLoop] = None,
 ) -> None:
     """Run an app locally"""
-    if loop is None:
-        loop = asyncio.new_event_loop()
-    loop.set_debug(debug)
-
-    # Configure if and only if in debugging mode and using the default logger
-    if loop.get_debug() and access_log and access_log.name == "aiohttp.access":
-        if access_log.level == logging.NOTSET:
-            access_log.setLevel(logging.DEBUG)
-        if not access_log.hasHandlers():
-            access_log.addHandler(logging.StreamHandler())
-
-    main_task = loop.create_task(
-        _run_app(
+    if loop is not None:
+
+        def loop_factory():
+            return loop
+
+    else:
+        loop_factory = events.get_running_loop
```

Comment:
I think loop_factory should default to None (maybe this is the cause of the RuntimeError above?).
```suggestion
    loop_factory = None if loop is None else lambda: loop
```